packages/components/components/elvis-stepper/src/react/utils.ts

```diff
@@ -43,25 +45,47 @@ export const numberShouldBeVisible = (
 };
 
 export const generateStatusMessage = (currentStep: number, steps: StepStates, errorSteps: number[]) => {
-  let statusMessage = `På steg ${currentStep}. `;
+  const lang = useLanguage();
+
+  const getTranslatedLabels = () => {
+    return {
+      onStep: lang === 'no' ? 'På steg' : 'On step',
+      previousStep: lang === 'no' ? 'Det forrige steget var' : 'The previous step was',
+      steps: lang === 'no' ? 'Steg' : 'Steps',
+      and: lang === 'no' ? 'og' : 'and',
+      wasInvalid: lang === 'no' ? 'var ugyldig' : 'was invalid',
+      statusLabel: (isCompleted: boolean) =>
+        isCompleted
+          ? lang === 'no'
+            ? 'vellykket'
+            : 'successful'
+          : lang === 'no'
+            ? 'ikke fullført'
+            : 'not completed',
+    };
+  };
```

Comment:
I would make this function return two objects; one in English, and one in Norwegian, then, you will also fix the... 

> Extract this nested ternary operation into an independent statement.

...warning that Sonar Cloud is giving.

Maybe something like this:

```
const getTranslatedLabels = () => {
    lang === 'no' ? {
      onStep: 'På steg',
      previousStep: 'Det forrige steget var',
      osv...
    } : {
      onStep: 'On step',
      previousStep: 'The previous step was',
    etc...
    }}
```

Reply:
This has been implemented.